### DISCUSSION TITLE
Does vsg currently support ray-tracing shadows and textures？

### DISCUSSION THREAD

**Discussion by WdsPjm** [vsg-dev/VulkanSceneGraph#902] [Q&A] [answered]
Hi !     I see a vsgraytrace.cpp demo. But I added a textured model and didn't render the texture properly. I was wondering if vsg currently supports ray-tracing shadows and textures

**Comment by robertosfield** [comment, marked as answer]
The ray tracing example is VERY simple, it doesn't attempt to illustrate anything more than the bare bones  use of the ray tracing extensions.

An more sophisticated example of ray tracing using the VSG can be found in the VulkanPBRT project:

https://github.com/Lachei/VulkanPBRT

I also the vsgopenmw project has use of the ray tracing:

https://github.com/vsgopenmw-dev/vsgopenmw
